### ISSUE TITLE
Full backup fails after timeout and restart

### ISSUE THREAD

**Issue by sfstar** [sabeechen/hassio-google-drive-backup#605]
<h1>Observed behaviour</h1>
My compressed full backups are around 15.6 GB in size and take longer than 4 hours to create.
It seems that after 4 hours the addon timeouts the backup creation after which the backup continues it's creation but gets ignored by the addon.
Resulting in an complete backup being present but not counting towards the chosen backup schema unless it's specifically marked as "do not ignore".
Thereby, triggering a 24/7 loop of creating new backups until storage runs out.

<h1>Expected behaviour</h1>
The backup not timing out unless it's creation gets stuck or the addon at least confirming whether an desired backup was created recently by the addon and matching it to the backup policy.

<h1>Possible work around</h1>
Disabling the ignoring of backups created outside of the addon.
This could stop the creation loop, but would include the partial backups created during upgrades (which I don't want to touch / automatically manage in case of issues arising with the specific upgraded component).

<h1>Disclaimers</h1>
I'm aware that a backup taking more than 4 hours might be less than desireable.
However, the backups are part of my multi tiered backup regime.
A zfs snapshot is created and synced to an offsite location and the google drive backup is a last resort backup in case the zfs backup mechanism doesn't work.
For which I accept the inconsistencies created by a long backup time between the different parts being backupped.

**Comment by sabeechen** [comment]
First off thanks for all the detail, I can tell you've been looking at this for a while and I imagine you've had a frustrating time waiting literally hours to see what goes wrong.  Having the detail from you saves a lot of time going back and forth trying to figure out your situation.

I actually intended for the addon to handle this situation (somewhat) gracefully.  What it _should_ be doing is noticing that the latest backup has the same name as the one it asked for a long time ago and them assume its the same one it asked for.

I have to do some of this tricky timeout handling and re-association because Home Assistant can't tell me information about a snapshot in progress.  So if a snapshot takes an unusual amount of time the addon can't reliably know if the last one just failed or if it should wait.

Everything I see in my code seems to be working correctly so long as the backup doesn't take more than 24 hours to complete.  So either I have a bug and my tests aren't catching it or your backups take more than 24 hours.  Do they?

**Comment by sabeechen** [comment]
I forgot to mention that you might be able to temporarily work around this situation until I can release or figure out a fix by opening up the addon's configuration in Home Assistant, editing it in "Yaml Mode" and adding the configuration option:
```yaml
pending_backup_timeout_seconds: 36000 # 36000 seconds == 10 hour
```

Though naturally you'll need to scale this up/down to however long your snapshots take plus a comfortable margin.  Once you add this the supervisor will complain about an invalid schema (its not a problem) and be careful not to save your configuration through the addon's web-ui because it will overwite the value.  This is an undocumented configuration option that I wouldn't recommend changing under other circumstances.

**Comment by sfstar** [comment]
Hello, thank you for your reply.
The backups currently take around 7.5 hours (based on there being about a 7.5 hour difference between when the backups creation started).
I will try the work around option you have given, but currently i'm 2 hours into the latest backup creation so verification whether this workaround has the desired effect might take a day

**Comment by sabeechen** [comment]
I think I found the problem, a little difficult to reproduce in my tests but it shows exactly the behavior you're seeing.  I was just doing some state handling out of order.  I'll release a fix soon.

As changing the supervisor supervisor, I totally agree it would be better if the supervisor exposed more information about backups in progress.  It would be especially useful to be able to stop a backup in progress.  Right now the only way to do that, AFAIK, is to reboot the machine.  For quite a while I've intended to make these changes in the supervisor but haven't gotten around to it.  I tend to focus more on what I can do in my little corner of code (the addon) :)

**Comment by sabeechen** [comment]
I just released v0.107.2 which includes a fix that should make long running backups like yours get recognized correctly by the addon.  The backups will still "time out" after 3 or 5 hours but after they finish (so long as its within 24 hours) the addon will remember that it created them, even if the addon gets restarted.  If you run into problems after updating, please feel free to reopen this issue or let me know.
